### PR TITLE
[BACKEND] Improve updates system

### PR DESCRIPTION
**Pull request by dragonDScript**
- Made the updates optional.
- Triggered the updates every 1 hour (changeable).

- [x] Add check for updates option to menu
- [x] ~~Make update interval changeable by the user.~~ Changeable in settings but not in the settings UI.
- [x] ~~Make the updates disableable by the user.~~ Changeable in settings but not in the settings UI.
- [x] Remind me later option.

I will work in the frontend in another PR.

### CODE REVIEW

File: src/background.js [hunk 1]
```diff
@@ -29,12 +29,14 @@ protocol.registerSchemesAsPrivileged([
   { scheme: 'app', privileges: { secure: true, standard: true } }
 ])
 
-const storage = new Store({
+global.storage = new Store({
```
Comment (lostdesign) [inline src/background.js:32]: haven't seen this before, is that a global object like window in the dom?

File: src/updater.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { dialog } from 'electron'
+import { autoUpdater } from 'electron-updater'
+
+var updatesAvailableCurrently
+
+// In ms
```
Comment (lostdesign) [inline src/updater.js:6]: What is that comment for?

File: src/updater.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    noLink: true
+}
+
+async function askForUpdates() {
+    if (!global.storage.get("enableUpdates")) return
+    const updateResult = await autoUpdater.checkForUpdates()
+    if (updateResult) {
+        const { response } = await dialog.showMessageBox(DIALOG_OPTS)
+        if (response === 1) { //ok button has been clicked
+            autoUpdater.downloadUpdate(updateResult.cancellationToken)
```
Comment (lostdesign) [inline src/updater.js:22]: Does this also install it once called?

**Comment by lostdesign** [comment]
Thanks for the PR, looks quite nice already. Ping me on discord if you need help or anything.

File: src/background.js (hunk 1 above)
Comment (dragonDScript) [reply src/background.js:32]: Yes, it's exactly like window.

File: src/updater.js (hunk 3 above)
Comment (dragonDScript) [reply src/updater.js:22]: I'm working on this.

File: src/updater.js (hunk 2 above)
Comment (dragonDScript) [reply src/updater.js:6]: Oh i forgot to remove it

**Comment by lostdesign** [comment]
Accepted and closing, as this now is going to be part of 1.4.0 - thanks.

**Pull request closed without merging**